internal/services/sources_service.go

```diff
@@ -6,44 +6,28 @@ import (
 	"net/http"
 
 	sources "github.com/RHEnVision/provisioning-backend/internal/clients/sources"
+	"github.com/RHEnVision/provisioning-backend/internal/headers"
 	"github.com/RHEnVision/provisioning-backend/internal/parsing"
 	"github.com/RHEnVision/provisioning-backend/internal/payloads"
 	"github.com/go-chi/chi/v5"
 	"github.com/go-chi/render"
 )
 
 func ListSources(w http.ResponseWriter, r *http.Request) {
-	client, err := sources.GetSourcesClient(r.Context())
+	client, err := sources.GetSourcesClientV2(r.Context())
 	if err != nil {
 		renderError(w, r, payloads.NewClientInitializationError(r.Context(), "sources client", err))
 		return
 	}
-	appTypeId, err := client.GetProvisioningTypeId(r.Context(), AddIdentityHeader)
-	if err != nil {
-		renderError(w, r, payloads.SourcesClientError(r.Context(), "get provisioning app type", err, 500))
-		return
-	}
-	resp, err := client.ListApplicationTypeSourcesWithResponse(r.Context(), appTypeId, &sources.ListApplicationTypeSourcesParams{}, AddIdentityHeader)
-	statusCode := resp.StatusCode()
-	if err != nil {
-		renderError(w, r, payloads.SourcesClientError(r.Context(), "list sources", err, statusCode))
-		return
-	}
-	if parsing.IsHTTPNotFound(statusCode) {
-		renderError(w, r, payloads.SourcesClientError(r.Context(), "sources not found", err, statusCode))
-		return
-	}
-	if !parsing.IsHTTPStatus2xx(statusCode) {
-		renderError(w, r, payloads.SourcesClientError(r.Context(), "sources client error", err, statusCode))
-		return
-	}
-	if err := render.RenderList(w, r, payloads.NewListSourcesResponse(resp.JSON200.Data)); err != nil {
+	sources, err := client.ListProvisioningSources(r.Context())
```

Comment:
@ezr-ondrej I need to add here error handling, should we use SourcesClientError? should I pass from ListProvisioningSources the statusCode as well? (we add SourcesClientError a status code parameter)

Reply:
I'd use SourcesClientError here, yes :+1: 
we can pass the status code, but it is not strictly necessary, if there is an error, we just return 500 with something like "hey there was an unexpected error in internal communication" ... so the internal statusCode is not too important here.

Reply:
but the error can be 404, we don't want to pass that if that is the case?

Reply:
well list should not return 404, right? it should return empty list. So even 404 is weird in this case

Reply:
```golang		
renderError(w, r, payloads.SourcesClientError(r.Context(), "sources client error", err, 500))
```
?

Reply:
looks sane to me :+1: